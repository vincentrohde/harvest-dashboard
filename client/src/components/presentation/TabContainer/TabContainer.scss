.TabContainer {
    width: 100%;
    margin-bottom: 20px;
    padding: 20px;
    background: #fff;
    border-radius: 10px;

    .title {
        margin: 0;
    }
}
