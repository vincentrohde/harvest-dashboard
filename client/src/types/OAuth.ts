export type oAuthType = {
    access_token: string;
    account_id: string;
    expiration_date: string;
    refresh_token: string;
};
