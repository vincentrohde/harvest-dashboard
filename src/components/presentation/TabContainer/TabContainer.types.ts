import { ReactNode } from 'react';

export interface TabContainerProps {
    children?: ReactNode;
    title?: string;
    caption?: string;
}
