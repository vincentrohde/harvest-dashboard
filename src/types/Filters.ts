export interface FiltersInterface {
    dateRange: string[] | 1;
}
