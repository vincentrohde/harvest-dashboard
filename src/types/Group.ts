export type groupsType = GroupInterface[];

export interface GroupInterface {
    id: number;
    name: string;
}