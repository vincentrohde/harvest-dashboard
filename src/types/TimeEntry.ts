import { ProjectInterface } from './Project';
import { TaskInterface } from './Task';

export type timeEntriesType = TimeEntryInterface[];

export interface ShallowTimeEntryInterface {
    hours: number | string;
    id: number;
    notes: string;
    spent_date: string;
    is_running: boolean;
    project: ProjectInterface;
    task: TaskInterface;
}

export interface TimeEntryInterface extends ShallowTimeEntryInterface {
    hours: number;
}

export interface ShallowSubmissionEntryInterface {
    notes: ShallowTimeEntryInterface['notes'];
    spent_date: ShallowTimeEntryInterface['spent_date'];
    hours: ShallowTimeEntryInterface['hours'];
    project_id: number | string;
    task_id: number | string;
}

export interface TimeEntrySubmissionInterface extends ShallowSubmissionEntryInterface {
    hours: TimeEntryInterface['hours'];
    project_id: number;
    task_id: number;
}