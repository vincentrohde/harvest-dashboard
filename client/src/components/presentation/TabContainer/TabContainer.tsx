// Libs
import React from 'react';

// Types
import {TabContainerProps} from './TabContainer.types';

// Style
import './TabContainer.scss';

const TabContainer = ({title, caption, children}: TabContainerProps) => (
    <div className="TabContainer">
        {(title || caption) && (
            <div className="header-container">
                {title && title.length > 0 && <h2 className="title">{title}</h2>}
                {caption && caption.length > 0 && <p className="tag">{caption}</p>}
            </div>
        )}
        {children && children}
    </div>
);

export default TabContainer;
