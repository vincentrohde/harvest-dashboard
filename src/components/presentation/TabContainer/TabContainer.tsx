// Libs
import React from 'react';

// Types
import {TabContainerProps} from './TabContainer.types';

// Style
import styles from './TabContainer.module.scss';

const TabContainer = ({title, caption, children}: TabContainerProps) => (
    <div className={styles.TabContainer}>
        {(title || caption) && (
            <div className={styles['header-container']}>
                {title && title.length > 0 && <h2 className={styles.title}>{title}</h2>}
                {caption && caption.length > 0 && <p className={styles.tag}>{caption}</p>}
            </div>
        )}
        {children && children}
    </div>
);

export default TabContainer;
