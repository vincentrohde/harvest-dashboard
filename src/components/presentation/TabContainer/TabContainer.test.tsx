import { render } from '@testing-library/react';
import React from 'react';
import '@testing-library/jest-dom/extend-expect';

// Components
import TabContainer from './TabContainer';

// Types
import { TabContainerProps } from './TabContainer.types';

const mockData: TabContainerProps = {
    title: 'My Test Container',
    caption: 'This is a rendered test container.'
}

describe('<TabContainer />', () => {
    it('renders', () => {
        const { asFragment } = render(
            <TabContainer {...mockData}>
                <p>Test</p>
            </TabContainer>
        );

        expect(asFragment()).toMatchSnapshot();
    });
})
