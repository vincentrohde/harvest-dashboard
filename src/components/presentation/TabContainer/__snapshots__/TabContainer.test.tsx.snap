// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabContainer /> renders 1`] = `
<DocumentFragment>
  <div
    class="TabContainer"
  >
    <div
      class="header-container"
    >
      <h2
        class="title"
      >
        My Test Container
      </h2>
      <p
        class="tag"
      >
        This is a rendered test container.
      </p>
    </div>
    <p>
      Test
    </p>
  </div>
</DocumentFragment>
`;
